cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/DefaultPluginContext.java

```diff
@@ -99,8 +99,19 @@ public PluginProperties getPluginProperties(String pluginId) {
 
   @Override
   public <T> T newPluginInstance(String pluginId, MacroEvaluator evaluator) throws InstantiationException {
```

Comment:
we should add unit tests for this method. You can look at ArtifactRepositoryTest to edit one of them or add unit tests there.